### PR TITLE
Refactor MethodConfigSpec

### PR DESCRIPTION
**Pull request by aweng98**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Clickable.scala [hunk 1]
```diff
@@ -25,14 +25,19 @@ trait Clickable extends LazyLogging { this: Component =>
...
-        // attempt to recover when this exact error occurred
+        // make an attempt to recover when this exact error occurred
         if (e.getMessage.contains("Other element would receive the click")) {
           logger.warn("retrying \"click on query\" after sleep 5 seconds")
           Thread.sleep(5000)
           click on query
         }
+    } finally {
+      Thread sleep 100
+      await notVisible cssSelector("[data-test-id=spinner]")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Clickable.scala]: a micro sleep then check for visibility of spinner.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Component.scala [hunk 2]
```diff
@@ -4,8 +4,6 @@ import com.typesafe.scalalogging.LazyLogging
 import org.broadinstitute.dsde.firecloud.FireCloudView
 import org.openqa.selenium.{JavascriptExecutor, WebDriver}
 
-import scala.util.Try
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Component.scala:7]: was not used.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/TabBar.scala [hunk 3]
```diff
@@ -7,7 +7,6 @@ case class TabBar(private val queryString: QueryString = TestId("tabs"))(implici
   def goToTab(name: String): Unit = {
     awaitReady()
     Link(s"$name-tab" inside this).doClick()
-    await notVisible cssSelector("[data-test-id=spinner]")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/TabBar.scala]: moved this check to doClick()

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/BaseFireCloudPage.scala [hunk 4]
```diff
@@ -8,17 +9,17 @@ import org.openqa.selenium.WebDriver
 abstract class BaseFireCloudPage(implicit webDriver: WebDriver) extends AuthenticatedPage {
 
   def goToWorkspaces(): WorkspaceListPage = {
-    click on testId("workspace-nav-link")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/BaseFireCloudPage.scala:11]: use Link.doClick

**Comment by aweng98** [comment]
To have two smaller test files:
1) Per QA OKRs, no single test file should be running longer than 10 minutes alone. A single large test file should be refactored into smaller test files, per test feature.
2) Use `withMethod` to create new method so tests cross contamination don't happen.
3) Run more tests in parallel that leverage multiple chrome nodes

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+import org.broadinstitute.dsde.firecloud.component.MessageModal
+import org.broadinstitute.dsde.firecloud.fixture.{TestData, UserFixtures}
+import org.broadinstitute.dsde.firecloud.page.workspaces.methodconfigs.{WorkspaceMethodConfigDetailsPage, WorkspaceMethodConfigListPage}
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.{Config, Credentials, UserPool}
+import org.broadinstitute.dsde.workbench.fixture._
+import org.broadinstitute.dsde.workbench.service.test.WebBrowserSpec
+import org.scalatest._
+
+class MethodRedactedSpec extends FreeSpec with Matchers with WebBrowserSpec with WorkspaceFixtures with UserFixtures with MethodFixtures with BillingFixtures {
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala:14]: Copied from MethodConfigSpec tests.

**Comment by aweng98** [comment]
round 1 tests passed

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/methodconfigs/ImportMethodConfigModal.scala [hunk 6]
```diff
@@ -105,5 +105,6 @@ private class ConfirmConfigCopyView(implicit webDriver: WebDriver) extends FireC
       nameField.setText(name.get)
 
     importButton.doClick()
+    awaitDismissed()
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/methodconfigs/ImportMethodConfigModal.scala]: modal should have gone away after `importButton.doClick`

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala [hunk 7]
```diff
@@ -80,7 +80,10 @@ class SubmissionDetailsPage(namespace: String, name: String, var submissionId: S
     */
   def waitUntilSubmissionCompletes(timeOut: FiniteDuration = 20.minutes): Unit = {
     retry[Boolean](5.seconds, timeOut) ({
-      goToMonitorTab().openSubmission(submissionId)
+      if (workflowStatusLabel.isVisible) {
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala:83]: clicking tab `Monitor` switches page between `Workflow` and `Submission`. `submissionId` is on `Submission` page, it's not on `Workflow` page.

**Comment by aweng98** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+              val list = await ready new WorkspaceMethodConfigListPage(billingProject, workspaceName)
+              list.hasConfig(methodConfigName) shouldBe false
+            }
+          }
+        }
+      }
+    }
+  }
+
+  "owner can abort a launched submission" in withWebDriver { implicit driver =>
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala]: This does not involve a redacted method, so it doesn't belong here.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+            submissionDetailsPage.abortSubmission()
+            submissionDetailsPage.waitUntilSubmissionCompletes()
+            submissionDetailsPage.getSubmissionStatus shouldBe submissionDetailsPage.ABORTED_STATUS
+          }
+        }
+      }
+    }
+  }
+
+  "import method config" - {
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala]: This does not involve a redacted method, so it doesn't belong here.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+  val methodConfigName: String = SimpleMethodConfig.configName + "_" + UUID.randomUUID().toString
+
+  "For a method config that references a redacted method" - {
+    "should be able to choose new method snapshot" in withWebDriver { implicit driver =>
+      val user: Credentials = UserPool.chooseProjectOwner
+      implicit val authToken: AuthToken = user.makeAuthToken()
+      withCleanBillingProject(user) { billingProject =>
+        withWorkspace(billingProject, "MethodRedactedSpec_choose_new_snapshot") { workspaceName =>
+          withMethod("MethodRedactedSpec_choose_new_snapshot", MethodData.SimpleMethod, 2) { methodName =>
+            api.methodConfigurations.createMethodConfigInWorkspace(billingProject, workspaceName, MethodData.SimpleMethod.copy(methodName = methodName),
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala]: if you add this here, you won't need to repeat the `copy` later:
`val method = MethodData.SimpleMethod.copy(methodName = methodName)`
and then you also won't need to reference `SimpleMethod` again.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    }
+
+    "launch analysis and delete errors on redacted method" in withWebDriver { implicit driver =>
+      val user = UserPool.chooseProjectOwner
+      implicit val authToken: AuthToken = user.makeAuthToken()
+      withCleanBillingProject(user) { billingProject =>
+        withWorkspace(billingProject, "MethodRedactedSpec_delete") { workspaceName =>
+          withMethod("MethodRedactedSpec_delete", MethodData.SimpleMethod, cleanUp = false) { methodName =>
+            api.methodConfigurations.createMethodConfigInWorkspace(billingProject, workspaceName, MethodData.SimpleMethod.copy(methodName = methodName),
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala]: same comment as above

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodConfigSpec.scala [hunk 12]
```diff
@@ -137,239 +128,52 @@ class MethodConfigSpec extends FreeSpec with Matchers with LazyLogging
...
-
-          withSignIn(user) { listPage =>
-            val methodConfigTab = listPage.enterWorkspace(billingProject, destWorkspaceName).goToMethodConfigTab()
 
-            val methodConfigDetailsPage = methodConfigTab.copyMethodConfigFromWorkspace(
-              billingProject, sourceWorkspaceName, method.methodNamespace, method.methodName)
-
-            methodConfigDetailsPage.isLoaded shouldBe true
-            methodConfigDetailsPage.methodConfigName shouldBe method.methodName
+        val method = MethodData.InputRequiredMethod
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodConfigSpec.scala]: Move this inside the `withMethod` block and `copy` the `methodName` in the same line.

(similar to my comment for MethodRedactedSpec)

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodLaunchSpec.scala [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+  val methodConfigName: String = SimpleMethodConfig.configName + "_" + UUID.randomUUID().toString
+  val wrongRootEntityErrorText: String = "Error: Method configuration expects an entity of type sample, but you gave us an entity of type participant."
+  val noExpressionErrorText: String = "Error: Method configuration expects an entity of type sample, but you gave us an entity of type sample_set."
+  val missingInputsErrorText: String = "is missing definitions for these inputs:"
+
+  "launch workflow and delete a workflow" in withWebDriver { implicit driver =>
+    val user = Config.Users.owner
+    implicit val authToken: AuthToken = user.makeAuthToken()
+    withCleanBillingProject(user) { billingProject =>
+      withWorkspace(billingProject, "TestSpec_FireCloud_launch_a_simple_workflow") { workspaceName =>
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodLaunchSpec.scala]: Update Workspace names.  Search for "TestSpec"

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodConfigSpec.scala [hunk 14]
```diff
@@ -1,375 +0,0 @@
...
-        val shouldUseCallCaching = false
-        api.importMetaData(billingProject, workspaceName, "entities", TestData.SingleParticipant.participantEntity)
-        api.methodConfigurations.createMethodConfigInWorkspace(billingProject, workspaceName,
-          MethodData.SimpleMethod, SimpleMethodConfig.configNamespace, methodName, 1,
-          SimpleMethodConfig.inputs, SimpleMethodConfig.outputs, MethodData.SimpleMethod.rootEntityType)
-
-        withSignIn(user) { _ =>
-          val methodConfigDetailsPage = new WorkspaceMethodConfigDetailsPage(billingProject, workspaceName, SimpleMethodConfig.configNamespace, methodName).open
-          val submissionDetailsPage = methodConfigDetailsPage.launchAnalysis(MethodData.SimpleMethod.rootEntityType, TestData.SingleParticipant.entityId, "", shouldUseCallCaching)
-          //TODO start the submission via API - reduce the amount of UI surface. - requires getting the submission ID
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodConfigSpec.scala:275]: put this back in please

**Review by jmthibault79 — APPROVED** [review]

**Comment by aweng98** [comment]
jenkins retest

**Pull request merged**